AEPAssurance/Source/Assurance.swift

```diff
@@ -243,26 +244,35 @@ public class Assurance: NSObject, Extension {
     /// If the timer get fired, then it shuts down the assurance extension.
     private func startShutDownTimer() {
         Log.debug(label: AssuranceConstants.LOG_TAG, "Assurance shutdown timer started. Waiting for 5 seconds to receive assurance session url.")
-        let queue = DispatchQueue.init(label: "com.adobe.assurance.shutdowntimer", qos: .background)
-        timer = createDispatchTimer(queue: queue, block: {
-            self.shutDownAssurance()
-        })
+        timerQueue.async { [weak self] in
+            guard let self else { return }
+            let queue = DispatchQueue.init(label: "com.adobe.assurance.shutdowntimer", qos: .background)
+            self.timer = self.createDispatchTimer(queue: queue, block: {
+                self.shutDownAssurance()
+            })
+        }
     }
 
     /// Shuts down the assurance extension by setting the `shouldProcessEvents` to false. On which no more events
     /// are listened by assurance extension
     /// @see readyForEvent
     private func shutDownAssurance() {
         Log.debug(label: AssuranceConstants.LOG_TAG, "Timeout - Assurance extension did not receive session url. Shutting down from processing any further events.")
-        invalidateTimer()
-        Log.debug(label: AssuranceConstants.LOG_TAG, "Clearing the queued events and purging Assurance shared state.")
-        sessionOrchestrator.terminateSession(purgeBuffer: true)
+        timerQueue.async { [weak self] in
```

Comment:
Since the `invalidateTimer` function is already guarded by the timerQueue, there’s no need to add additional synchronization in this method.